src/dodal/devices/bimorph_mirrors/CAENels_bimorph_mirror_interface.py

```diff
@@ -0,0 +1,273 @@
+    all_volt: EpicsSignal = Component(EpicsSignal, "ALLVOLT")
+    operation_mode_readback_value: EpicsSignalRO = Component(
+        EpicsSignalRO, "OPMODE_RBV"
+    )
+    # Basically just the number of channels:
+    channels: EpicsSignalRO = Component(EpicsSignalRO, "CHANNELS")
+    status: EpicsSignalRO = Component(EpicsSignalRO, "STATUS")
+    board_temperature: EpicsSignalRO = Component(EpicsSignalRO, "TEMPS")
+    # PV suffix RESETERR.PROC, ERR, might be confusing. Errors come through
+    # ..:BUSY kinda:
+    reset_alarms: EpicsSignal = Component(EpicsSignal, "RESETERR.PROC")
+    alarm_status: EpicsSignalRO = Component(EpicsSignalRO, "ERR")
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # Lists of channels for easy access:
+        self._voltage_target_channels: list = []
+        self._voltage_target_readback_value_channels: list = []
+        self._shift_channels: list = []
+        self._voltage_out_channels: list = []
+        self._voltage_out_readback_value_channels: list = []
+        self._status_channels: list = []
+
+        # because self.status is compiled in a signal at instantiation, these have to be define in __init__:
+        self.wait_till_idle = partial(
+            self.wait_for_signal_value,
+            self.status,
+            Status.IDLE,
+            signal_range=[Status.IDLE, Status.BUSY],
+        )
+        self.wait_till_busy = partial(
+            self.wait_for_signal_value,
+            self.status,
+            Status.BUSY,
+            signal_range=[Status.IDLE, Status.BUSY],
+        )
+
+    def wait_for_signal_value(
+        self,
+        signal: EpicsSignal,
+        value: Any,
+        timeout: float = 10.0,
+        sleep_time: float = 0.1,
+        signal_range: Union[list, None] = None,
+    ):
+        """Wait for a signal to display given value. Default to polling time of 0.1 seconds, and timeout after 10.0.
+        If signal_range is non-none, an exception will be raised if the signal value does not exist in signal_range list.
+
+        Args:
+            signal: The signal to be read from
+            value: Once this value is read from signal, the function will return
+            timeout: Time before an exception is raised in seconds (default 10.0)
+            sleep_time: Polling time in seconds (default 0.1)
+            signal_range: Range of allowed values. If value outside of this is read, an exception will be thrown
+        """
+        stamp = time.time()
+        res = signal.read()[signal.name]["value"]
+
+        while res != value:
+            if signal_range is not None:
+                if res not in signal_range:
+                    raise Exception(
+                        f"Out of range: {signal} showing {res} not in {signal_range}"
+                    )
+
+            if time.time() - stamp > timeout:
+                raise Exception(f"Timeout waiting for {signal} to show {value}")
```

Comment:
Must: We shouldn't be raising an exception here for every iteration of this loop.